tests/test_stall_epready.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+# Copyright 2016-2021 XMOS LIMITED.
+# This Software is subject to the terms of the XMOS Public Licence: Version 1.
+
+# Directed test for (github) issue #58
+
+import xmostest
+from usb_packet import *
+import usb_packet
+from helpers import do_usb_test, RunUsbTest
+from usb_session import UsbSession
+from usb_transaction import UsbTransaction
+
+
+def do_test(arch, clk, phy, usb_speed, seed, verbose=False):
+
+    address = 1
+    pktLength = 10
+
+    session = UsbSession(
+        bus_speed=usb_speed, run_enumeration=False, device_address=address
+    )
+
+    ep_ctrl = 2
+    ep = 1
+
+    # Expect test EP's to be halted
+    session.add_event(
+        UsbTransaction(
+            session,
+            deviceAddress=address,
+            endpointNumber=ep,
+            endpointType="BULK",
+            direction="OUT",
+            dataLength=pktLength,
+            halted=True,
+        )
+    )
+
+    session.add_event(
+        UsbTransaction(
+            session,
+            deviceAddress=address,
+            endpointNumber=ep,
+            endpointType="BULK",
+            direction="OUT",
+            dataLength=pktLength,
+            halted=True,
+        )
+    )
```

Comment:
As above, we're sending the same packet twice. Just checking this isn't a cut 'n' paste error ;-)